rfcs/RFC-Uploads.md

```diff
@@ -0,0 +1,348 @@
+Figure 1: The Upload Design
+
+### Data Portal Browser App
+
+The DP Browser App will be used authenticate the user, and interact with the DP Backend to upload files.
+It will also use AWS JS SDK to upload files from local storage to AWS. These feature will extend the existing DP
+Frontend.
+
+### Data Portal Backend
+
+The DP Backend is responsible for handling users requests. It will generated presigned URL and access tokens
+for the DP Browser App to facilitate the uploading of files. It will also enqueue shared links for downloading.
+This additional functionality can be built into the existing DP Backend Lambda.
+
+### Download Queue
+
+The Download Queue is an [AWS SQS](https://aws.amazon.com/sqs/) that will track pending and in progress download jobs.
+
+#### Download Job Entry
+
+These are the fields that will be in the download job placed in the queue:
+
+- Link - a to the link to download. Use to retrieve the file from CSP.
+- collect_id - identifies the collection. Used to determine the storage location.
+- dataset_id - identifies the dataset. Used to determine the storage location
+- asset_id - identifies the asset. Used to determine the storage location, and file name.
+- file size - the size of the asset to download. Used to determine the download method, and EC2 instance size.
+
+### Data Portal APIs
+
+The DP Browser App will use these endpoint to interact with the Backend using these APIs. These are the new
+API needs to complete an upload.
+
+#### POST {collection_id}/{data set_id}/upload/link
+
+An authenticated user can send a request to upload a file from a shared link to their data set in their collection.
+
+##### Request
+
+- Link - a shared link to the file
+- collection_id - identifies the collection
+- data set_id - identifies the data set
+
+##### Response
+
+- Asset_uuid - identifies the name of the new file created.
+
+#### POST {collection_id}/{dataset_id}/upload/file
+
+An authenticated user can send a request to upload a file directly from their local machine to their data set in their
+collection.
+
+##### Request: POST {collection_id}/{dataset_id}/upload/file
+
+- File size - the size of the file to be uploaded
+- collection_id - identifies the collection
+- data set_id - identifies the data set
+
+##### Response: POST {collection_id}/{dataset_id}/upload/file
+
+- Access Token - allows the user to upload their file to S3 in multiple parts.
+- Presigned URL - allows the user to upload their file to S3 in a single part.
+- asset_uuid - identifies the name of the new file created.
+
+#### POST /dp/v1/dataset/{dataset_uuid}/asset/{asset_uuid}/upload
+
+Used to request an access token to continue uploading a file to S3 by an authenticated user who owns the data set. An
+error is returned if the file upload has been completed already.
+
+##### Request: POST /dp/v1/dataset/{dataset_uuid}/asset/{asset_uuid}/upload
+
```

Comment:
Due the lack of narrative for the flow, I'm confused when this would need to be called since the POST above also returns an Access Token? And how is the Access Token used?

Reply:
This is pretty much a duplicate of the previous endpoint. I think we can merge the functionality.

Reply:
This endpoint has been removed